src/sentry/static/sentry/app/views/issueList/actions/actionSet.tsx

```diff
@@ -0,0 +1,316 @@
+
+import ActionLink from 'app/components/actions/actionLink';
+import IgnoreActions from 'app/components/actions/ignore';
+import DropdownLink from 'app/components/dropdownLink';
+import MenuItem from 'app/components/menuItem';
+import Tooltip from 'app/components/tooltip';
+import {IconEllipsis, IconIssues, IconPause, IconPlay} from 'app/icons';
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+import {Organization, Project, ResolutionStatus} from 'app/types';
+import Projects from 'app/utils/projects';
+
+import ResolveActions from './resolveActions';
+import {ConfirmAction, getConfirm, getLabel} from './utils';
+
+type Props = {
+  orgSlug: Organization['slug'];
+  queryCount: number;
+  query: string;
+  realtimeActive: boolean;
+  allInQuerySelected: boolean;
+  anySelected: boolean;
+  multiSelected: boolean;
+  issues: Set<string>;
+  onShouldConfirm: (action: ConfirmAction) => boolean;
+  onDelete: () => void;
+  onRealtimeChange: () => void;
+  onMerge: () => void;
+  onUpdate: (data?: any) => void;
+  selectedProjectSlug?: string;
+  hasInbox?: boolean;
+};
+
+function ActionSet({
+  orgSlug,
+  queryCount,
+  query,
+  realtimeActive,
+  allInQuerySelected,
+  anySelected,
+  multiSelected,
+  issues,
+  onUpdate,
+  onShouldConfirm,
+  onDelete,
+  onRealtimeChange,
+  onMerge,
+  selectedProjectSlug,
+  hasInbox,
+}: Props) {
+  const numIssues = issues.size;
+  const confirm = getConfirm(numIssues, allInQuerySelected, query, queryCount);
+  const label = getLabel(numIssues, allInQuerySelected);
+
+  // merges require a single project to be active in an org context
+  // selectedProjectSlug is null when 0 or >1 projects are selected.
+  const mergeDisabled = !(multiSelected && selectedProjectSlug);
+
+  return (
+    <Wrapper hasInbox={hasInbox}>
+      {hasInbox && (
+        <div className="btn-group hidden-sm hidden-xs">
+          <StyledActionLink
+            className="btn btn-default btn-sm action-merge"
+            data-test-id="button-acknowledge"
+            disabled={!anySelected}
+            onAction={() => onUpdate({inbox: false})}
+            shouldConfirm={onShouldConfirm(ConfirmAction.ACKNOWLEDGE)}
+            message={confirm(ConfirmAction.MARK, false, ' as reviewed')}
+            confirmLabel={label('Mark', ' as reviewed')}
+            title={t('Mark Reviewed')}
+          >
+            <StyledIconIssues size="xs" />
+            {t('Mark Reviewed')}
+          </StyledActionLink>
+        </div>
+      )}
+      {selectedProjectSlug ? (
+        <Projects orgId={orgSlug} slugs={[selectedProjectSlug]}>
+          {({projects, initiallyLoaded, fetchError}) => {
+            const selectedProject = projects[0];
+            return (
+              <ResolveActions
+                onShouldConfirm={onShouldConfirm}
+                onUpdate={onUpdate}
+                anySelected={anySelected}
+                orgSlug={orgSlug}
+                params={{
```

Comment:
Why did you go with a params object instead of props?

Reply:
I could have changed it to prop but I didn't want to change it too much from the original so I decided to leave it as params
https://github.com/getsentry/sentry/blob/073bfadcf094943e793df4f6f7b768e3b15a1fee/src/sentry/static/sentry/app/views/issueList/actions.tsx#L364